lib/spack/spack/test/conftest.py

```diff
@@ -49,9 +48,62 @@
 from spack.version import Version
 
 
-##########
-# Monkey-patching that is applied to all tests
-##########
+#
+# These fixtures are applied to all tests
+#
+@pytest.fixture(scope='function', autouse=True)
```

Comment:
Just for information. If using these fixtures even where it's not needed becomes a problem, we can disable them selectively with just a few modifications:
```python
# In this file
@pytest.fixture(scope='function', autouse=True)
def no_chdir(request):
    if 'disable_no_chdir' in request.keywords:
        return 

    # Follows the implementation above

# In the file where the test resides
@pytest.mark.disable_no_chdir
def test_foo(...):
    ...
```

Reply:
That might be cleaner than destroying the stages when installs intentionally fail.  I’ll add a keep_stage keyword.

Reply:
This is addressed in the updated commits.